src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -345,15 +395,34 @@ public void OnResponseAbort(Exception abortException)
                 }
             }
 
-            public async Task ReadResponseHeadersAsync()
+            public async Task ReadResponseHeadersAsync(TaskCompletionSource<bool> tcs = null)
```

Comment:
Please rename `tcs` to something more meaningful, i.e. what does this represent / when should it be completed?